src/verify_cert.rs

```diff
@@ -14,14 +14,44 @@
 
 use crate::{
     cert::{self, Cert, EndEntityOrCa},
-    der, signed_data, subject_name, time, Error, SignatureAlgorithm, TrustAnchor,
+    der, signed_data, subject_name, time, CertRevocationList, Error, SignatureAlgorithm,
+    TrustAnchor,
 };
 
+/// Specifies how much of a certificate chain is checked for revocation status.
+#[derive(Debug, PartialEq, Eq)]
+pub enum RevocationCheckDepth {
+    /// Only check the end entity certificate for revocation status.
+    EndEntity,
+    /// Check both the end entity certificate as well as any relevant intermediate certificates
+    /// for revocation status.
+    Chain,
+}
```

Comment:
This requires exporting `EndEntityOrCa` but otherwise worked OK.

I lifted the depth enum out of the `wbepki` code and into the integration test driver `.rs` that's used by the generated test cases. I did this half to provide an example in case it was interesting for consumers, and half to keep the existing tests working more or less as-is. I'm also happy to simplify the tests to not handle EE vs CA distinctly.